### PR TITLE
max usage count fixes

### LINKED ISSUE #5741: Dynamically changing Pool.maxUsageCount may not work if the usage count has overflowed
**Jetty version**
9.4.x, 10.0.x, 11.0.x

**Description**
`org.eclipse.jetty.util.Pool` has a `maxUsageCount` property that can dynamically be changed. This is just an indicator of what the max usage count can be, it isn't the actual counter; the latter is in the `Entry` inner class.

But `Entry.tryAcquire` blindly increments its counter without checking for overflow, so if an entry's counter overflowed before the  `maxUsageCount` was set, then the entry is going to be reused again potentially for billions of times before its counter becomes positive again as the check is implemented as `if (usageCount >= maxUsageCount) do_not_reuse();`

### PR DESCRIPTION
**Pull request by lorban**
Closes #5741

### CODE REVIEW

File: jetty-util/src/main/java/org/eclipse/jetty/util/Pool.java [hunk 1]
```diff
@@ -166,16 +166,42 @@ public final void setMaxMultiplex(int maxMultiplex)
...
+        {
+            copy = new ArrayList<>(entries);
+        }
+
+        // Iterate the copy and close overused entries.
+        for (Entry entry : copy)
+        {
+            if (entry.isIdleAndOverUsed() && remove(entry) && entry.pooled instanceof Closeable)
+                IO.close((Closeable)entry.pooled);
+        }
```
Comment (gregw) [inline jetty-util/src/main/java/org/eclipse/jetty/util/Pool.java]: Would it not be better to do all the removals whilst the lock is held and only do the closing after releasing the lock:

```suggestion
        List<Entry> overused = new ArrayList();
        try (Locker.Lock l = locker.lock())
        {
            // look for overused here and if found remove from entries and add to overused
        }

        // Iterate the copy and close overused entries.
        for (Entry entry : overused)
        {
            if (entry.pooled instanceof Closeable)
                IO.close((Closeable)entry.pooled);
        }
```

**Review by gregw — CHANGES_REQUESTED** [review]
I like the clean in the setter, but I think it can be done better with a single grab of the lock.
But if that is too complex, I'd accept the current impl... but only after you try doing it with one lock grab.

File: jetty-util/src/main/java/org/eclipse/jetty/util/Pool.java (hunk 1 above)
Comment (lorban) [reply jetty-util/src/main/java/org/eclipse/jetty/util/Pool.java]: Good idea, that can even be improved a bit further. Let me get back to this.

**Comment by lorban** [comment]
> I like the clean in the setter, but I think it can be done better with a single grab of the lock.
> But if that is too complex, I'd accept the current impl... but only after you try doing it with one lock grab.

What do you mean with "_with a single grab of the lock_"?

**Review by gregw — APPROVED** [review]

**Pull request merged**